packages/core/src/memory/memory.ts

```diff
@@ -130,14 +155,64 @@ export abstract class MastraMemory extends MastraBase {
     return this._storage;
   }
 
+  /**
+   * Sets the storage provider for this memory instance.
+   * Automatically wraps the storage with initialization support.
+   *
+   * @param storage - The storage provider to use
+   * @internal
+   *
+   * @example
+   * ```typescript
+   * import { PgStorage } from "@mastra/pg";
+   *
+   * const storage = new PgStorage({
+   *   connectionString: "postgresql://..."
+   * });
+   * memory.setStorage(storage);
+   * ```
+   */
```

Comment:
I think some of these methods are used by us but shouldn't be recommended to users in reference docs. For example users shouldn't call this - which of course means this class shouldn't have it or it should be something like `__setStorage`.

Memory should always be configured via constructor config or in stream/generate args